internal/webhooks/cluster.go

```diff
@@ -262,8 +262,12 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 	}
 
 	if oldCluster != nil { // On update
-		// Topology or Class can not be added on update.
+		// Topology or Class can not be added on update unless manual adaption annotation is set
 		if oldCluster.Spec.Topology == nil || oldCluster.Spec.Topology.Class == "" {
+			if clusterv1.ClusterTopologyAdoptionTypeManual == newCluster.Annotations[clusterv1.ClusterTopologyAdoptionAnnotation] {
```

Comment:
I wonder if this skips too much. As soon as we hit this if branch the entire subsequent validation 283++ is ignored